snuba/subscriptions/data.py

```diff
@@ -166,10 +180,71 @@ class SnQLSubscriptionData(SubscriptionData):
     def build_request(
         self, dataset: Dataset, timestamp: datetime, offset: Optional[int], timer: Timer
     ) -> Request:
-        # TODO: Parse the query without doing the full validation
-        # TODO: Add time range conditions
-        # TODO: add offset condition
-        raise NotImplementedError
+        schema = RequestSchema.build_with_extensions(
+            {}, SubscriptionRequestSettings, Language.SNQL,
+        )
+
+        def add_conditions(query: Union[CompositeQuery[Entity], Query]) -> None:
+            # TODO: Support composite queries with multiple entities.
+            from_clause = query.get_from_clause()
+            if not isinstance(from_clause, Entity):
+                raise InvalidSubscriptionError("Only simple queries are supported")
+            entity = get_entity(from_clause.key)
+            required_timestamp_column = entity.required_time_column
+            if required_timestamp_column is None:
+                raise InvalidSubscriptionError(
+                    "Entity must have a timestamp column for subscriptions"
+                )
+
+            conditions_to_add: List[Expression] = [
```

Comment:
This assumes the entity we are running a subscription on has a project_column (like the legacy code). I think we should abstract the conditions to be generated at every subscription out of the generic subscription data class. This is true for both legacy and new implementation. The legacy one was built before entities so there was no other option, now it should be easier.